core/src/main/java/apoc/trigger/TriggerHandler.java

```diff
@@ -283,6 +284,10 @@ private boolean when(Map<String, Object> selector, Phase phase) {
 
     @Override
     public void start() throws Exception {
+        if (apocConfig.getBoolean(TRIGGER_PREVENT_PERSIST, false)) {
```

Comment:
potresti fare un metodo ifPersistDo:
```java
public void ifPersistDo(Supplier<Void> supplier) {
  if (!apocConfig.getBoolean(TRIGGER_PERSIST, false)) {
    removeAll();
    return;
  }
  supplier.get();

}   
```

e poi:

```java
public void start() throws Exception {
  ifPersistDo(() -> {
    updateCache();
    long refreshInterval = apocConfig().getInt(TRIGGER_REFRESH, 60000);
    restoreTriggerHandler = jobScheduler.scheduleRecurring(Group.STORAGE_MAINTENANCE, () -> {
      if (getLastUpdate() > lastUpdate) {
        updateCache();
      }
     }, refreshInterval, refreshInterval, TimeUnit.MILLISECONDS);
  });
}
```

poi `ifPersistDo` lo puoi riusare anche sotto in `stop`